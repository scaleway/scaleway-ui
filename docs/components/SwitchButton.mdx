---
name: SwitchButton
menu: Components
---

import { Playground } from 'docz'
import { Grid, Col, Row, SwitchButton, BorderedBox } from 'scaleway-ui'
import ControlValue from 'docs/utils/ControlValue'


# SwitchButton

## Basic

`SwitchButton` uses a `input` of type "radio" and they can only be used as controlled input.

<Playground>
  <ControlValue value="a">
    {({ value, onChange }) => (
      <Grid>
        <Row>
          <Col xs={3}>
            <SwitchButton
              width={1}
              name="switch"
              value="a"
              checked={value === 'a'}
              onChange={e => onChange(e.currentTarget.value)}
            >
              This is A
            </SwitchButton>
          </Col>
          <Col xs={3}>
            <SwitchButton
              width={1}
              name="switch"
              value="b"
              checked={value === 'b'}
              onChange={e => onChange(e.currentTarget.value)}
            >
              This is B
            </SwitchButton>
          </Col>
        </Row>
      </Grid>
    )}
  </ControlValue>
</Playground>

## Variant

Use `variant` prop to change style. Available variants: `default`, `segment`.

Check code example to reproduce exactly `segment` variant.

<Playground>
  <ControlValue value="a">
    {({ value, onChange }) => (
      <Grid>
        <Row>
          <Col xs={3}>
            <SwitchButton
              width={1}
              name="switch"
              value="a"
              variant="default"
              checked={value === 'a'}
              onChange={e => onChange(e.currentTarget.value)}
            >
              This is A
            </SwitchButton>
          </Col>
          <Col xs={3}>
            <SwitchButton
              width={1}
              name="switch"
              value="b"
              variant="default"
              checked={value === 'b'}
              onChange={e => onChange(e.currentTarget.value)}
            >
              This is B
            </SwitchButton>
          </Col>
        </Row>
      </Grid>
    )}
  </ControlValue>
  <br/>
  <ControlValue value="a">
    {({ value, onChange }) => (
      <Grid>
        <Row width={320}>
          <Col pr={0}>
            <BorderedBox p="4px">
              <Row>
                <Col xs={6} pr={0}>
                  <SwitchButton
                    width={1}
                    name="switch"
                    value="a"
                    variant='segment'
                    checked={value === 'a'}
                    onChange={e => onChange(e.currentTarget.value)}
                  >
                    This is A
                  </SwitchButton>
                </Col>
                <Col xs={6} pl={0}>
                  <SwitchButton
                    width={1}
                    name="switch"
                    value="b"
                    variant='segment'
                    checked={value === 'b'}
                    onChange={e => onChange(e.currentTarget.value)}
                  >
                    This is B
                  </SwitchButton>
                </Col>
              </Row>
            </BorderedBox>
          </Col>
        </Row>
      </Grid>
    )}
  </ControlValue>
</Playground>

## Disabled

Use `disabled` prop to disable one or both `SwitchButton`.

<Playground>
  <ControlValue value="a">
    {({ value, onChange }) => (
      <Grid>
        <Row>
          <Col xs={3}>
            <SwitchButton
              width={1}
              name="switch"
              value="a"
              checked={value === 'a'}
              onChange={e => onChange(e.currentTarget.value)}
            >
              This is A
            </SwitchButton>
          </Col>
          <Col xs={3}>
            <SwitchButton
              width={1}
              name="switch"
              value="b"
              disabled
              checked={value === 'b'}
              onChange={e => onChange(e.currentTarget.value)}
            >
              This is B
            </SwitchButton>
          </Col>
        </Row>
      </Grid>
    )}
  </ControlValue>
</Playground>
