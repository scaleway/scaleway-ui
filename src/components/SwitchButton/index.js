import { css } from '@emotion/core'
import { SwitchState, Box } from '@smooth-ui/core-em'
import PropTypes from 'prop-types'
import React from 'react'
import { theme } from '../../theme'
import { Tooltip } from '../Tooltip'

const styles = {
  container: ({ checked, disabled }) =>
    css({
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'center',
      borderRadius: 4,
      alignItems: 'center',
      borderStyle: 'solid',
      borderColor: theme.gray350,
      borderWidth: 1,
      padding: 16,
      transition: 'color 0.2s, border-color 0.2s, box-shadow 0.2s',
      userSelect: 'none',
      touchAction: 'manipulation',
      color: theme.gray550,
      fontSize: 16,
      lineHeight: '22px',
      position: 'relative',
      fontWeight: 500,
      input: {
        cursor: disabled ? 'not-allowed' : 'pointer',
      },
      ':hover,:focus': !checked &&
        !disabled && {
          color: theme.gray550,
          borderColor: theme.primary,
          boxShadow: `0 0 8px 2px ${theme.gray200}`,
          select: {
            color: theme.gray550,
          },
        },
    }),
  focused: css({
    boxShadow: `0 0 8px 2px ${theme.gray200}`,
  }),
  checked: css({
    cursor: 'auto',
    color: theme.primary,
    borderColor: theme.primary,
  }),
  disabled: css({
    cursor: 'not-allowed',
    color: theme.gray350,
    backgroundColor: theme.gray50,
    borderColor: theme.gray350,
    pointerEvents: 'none',
  }),
}

const variants = {
  segment: {
    switchButton: css`
      font-size: 14;
      height: 40px;
      transition: none;
      :hover,
      :focus {
        box-shadow: none;
        border: none;
        color: ${theme.gray700};
      }
      border-radius: 4px;
      box-shadow: none;
    `,
    checked: css`
      background-color: ${theme.primary};
      color: ${theme.white};
      :hover {
        color: ${theme.white};
      }
    `,
    notChecked: css`
      background-color: white;
      color: ${theme.gray700};
      border-color: ${theme.transparent};
      :hover,
      :focus {
        color: ${theme.gray700};
        border: none;
        box-shadow: none;
      }
    `,
  },
}

export const SwitchButton = ({
  checked,
  disabled,
  defaultChecked,
  onChange,
  onFocus,
  onBlur,
  name,
  value,
  size,
  children,
  tooltip,
  variant,
  ...props
}) => (
  <SwitchState
    name={name}
    value={value}
    checked={checked}
    defaultChecked={defaultChecked}
    onChange={onChange}
    onFocus={onFocus}
    onBlur={onBlur}
    disabled={disabled}
  >
    {({ focused, checked, disabled, input }) => (
      <Tooltip text={tooltip}>
        <Box
          as="label"
          htmlFor={`${name}-${value}`}
          disabled={disabled}
          css={[
            styles.container({ checked, disabled }),
            focused && styles.focused,
            checked && styles.checked,
            disabled && styles.disabled,
            variant && variants[variant]?.switchButton,
            checked
              ? variants[variant]?.checked
              : variants[variant]?.notChecked,
          ]}
          {...props}
        >
          {typeof children === 'function'
            ? children({ focused, checked, disabled })
            : children}
          <input
            id={`${name}-${value}`}
            type="radio"
            {...input}
            style={{ ...input.style, ...(checked ? { zIndex: -1 } : null) }}
          />
        </Box>
      </Tooltip>
    )}
  </SwitchState>
)

SwitchButton.defaultProps = {
  size: 24,
  disabled: false,
  defaultChecked: false,
  variant: 'default',
  onFocus: () => {},
  onBlur: () => {},
}

SwitchButton.propTypes = {
  checked: PropTypes.bool.isRequired,
  disabled: PropTypes.bool,
  defaultChecked: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
  onFocus: PropTypes.func,
  onBlur: PropTypes.func,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  size: PropTypes.number,
  variant: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
}
