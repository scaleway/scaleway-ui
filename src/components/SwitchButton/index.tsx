import { Theme, css } from '@emotion/react'
import styled from '@emotion/styled'
import { transparentize } from 'polished'
import PropTypes from 'prop-types'
import React, {
  ChangeEventHandler,
  FocusEventHandler,
  FunctionComponent,
  LabelHTMLAttributes,
  ReactNode,
} from 'react'
import { Radio } from 'reakit'
import Box from '../Box'
import Tooltip from '../Tooltip'

const variants = {
  segment: ({ theme }: { theme: Theme }) => css`
    font-size: 14;
    height: 40px;
    transition: none;
    border-radius: 4px;
    box-shadow: none;

    &:hover,
    &:focus {
      box-shadow: none;
      border-color: ${theme.colors.transparent};
      color: ${theme.colors.gray700};
    }

    &[aria-checked='true'] {
      background-color: ${theme.colors.primary};
      color: ${theme.colors.white};
      :hover {
        color: ${theme.colors.white};
      }
    }

    &[aria-checked='false'] {
      background-color: white;
      color: ${theme.colors.gray700};
      border-color: ${theme.colors.transparent};
      :hover,
      :focus {
        color: ${theme.colors.gray700};
        border-color: ${theme.colors.transparent};
        box-shadow: none;
      }
    }
  `,
} as const

type Variants = keyof typeof variants

const switchButtonVariants = Object.keys(variants) as Variants[]

const active = ({ theme }: { theme: Theme }) => css`
  &:hover,
  &:focus {
    color: ${theme.colors.gray550};
    border-color: ${theme.colors.primary};
  }

  &:hover {
    box-shadow: 0 0 8px 2px ${theme.colors.gray200};
  }

  &:focus {
    box-shadow: 0 0 1px 2px ${transparentize(0.75, theme.colors.primary)};
  }
`

const disabledClass = ({ theme }: { theme: Theme }) => css`
  cursor: not-allowed;
  color: ${theme.colors.gray350};
  background-color: ${theme.colors.gray50};
  border-color: ${theme.colors.gray350};
  pointer-events: none;

  &[aria-checked='true'] {
    color: ${theme.colors.gray350};
    border-color: ${theme.colors.gray350};
  }
`

type StyledSwitchProps = {
  checked?: boolean
  disabled?: boolean
  variant?: Variants
} & LabelHTMLAttributes<HTMLLabelElement>

const StyledSwitch = styled(Box, {
  shouldForwardProp: prop => !['variant'].includes(prop.toString()),
})<StyledSwitchProps>`
  display: flex;
  flex-direction: row;
  justify-content: center;
  border-radius: 4px;
  align-items: center;
  border-style: solid;
  border-color: ${({ theme }) => theme.colors.gray350};
  border-width: 1px;
  padding: 16px;
  transition: color 0.2s, border-color 0.2s, box-shadow 0.2s;
  user-select: none;
  touch-action: manipulation;
  color: ${({ theme }) => theme.colors.gray550};
  font-size: 16px;
  line-height: 22px;
  position: relative;
  font-weight: 500;
  cursor: pointer;

  &[aria-checked='true'] {
    cursor: auto;
    color: ${({ theme }) => theme.colors.primary};
    border-color: ${({ theme }) => theme.colors.primary};
  }

  ${({ checked, disabled }) => !checked && !disabled && active}
  ${({ disabled }) => disabled && disabledClass}
  ${({ variant }) => variant && variants[variant]}
`

const StyledRadio = styled(Radio)`
  position: absolute;
  opacity: 0.01;
`

type SwitchButtonProps = StyledSwitchProps & {
  children:
    | ReactNode
    | (({
        checked,
        disabled,
      }: {
        checked: boolean
        disabled: boolean
      }) => JSX.Element)
  name: string
  onBlur?: FocusEventHandler
  onChange: ChangeEventHandler
  onFocus?: FocusEventHandler
  tooltip?: string
  value: string
}

const SwitchButton: FunctionComponent<SwitchButtonProps> = ({
  checked = false,
  disabled = false,
  onChange,
  onFocus,
  onBlur,
  name,
  value,
  children,
  tooltip,
  variant,
  ...props
}) => (
  <Tooltip text={tooltip}>
    <StyledSwitch
      as="label"
      htmlFor={`${name}-${value}`}
      disabled={disabled}
      variant={variant}
      checked={checked}
      aria-checked={checked}
      {...props}
    >
      {typeof children === 'function'
        ? children({ checked, disabled })
        : children}
      <StyledRadio
        checked={checked}
        disabled={disabled}
        id={`${name}-${value}`}
        name={name}
        onBlur={onBlur}
        onChange={onChange}
        onFocus={onFocus}
        type="radio"
        value={value}
      />
    </StyledSwitch>
  </Tooltip>
)

SwitchButton.propTypes = {
  checked: PropTypes.bool,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.func,
  ]).isRequired,
  disabled: PropTypes.bool,
  name: PropTypes.string.isRequired,
  onBlur: PropTypes.func,
  onChange: PropTypes.func.isRequired,
  onFocus: PropTypes.func,
  tooltip: PropTypes.string,
  value: PropTypes.string.isRequired,
  variant: PropTypes.oneOf(switchButtonVariants),
}

export default SwitchButton
