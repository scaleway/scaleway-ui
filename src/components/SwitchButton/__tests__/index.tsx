import userEvent from '@testing-library/user-event'
import SwitchButton from '..'
import {
  renderWithTheme,
  shouldMatchEmotionSnapshot,
} from '../../../helpers/jestHelpers'

describe('SwitchButton', () => {
  test('renders correctly', () =>
    shouldMatchEmotionSnapshot(
      <SwitchButton
        name="test"
        onChange={() => {}}
        leftButton={{
          label: 'Left',
          value: 'left',
        }}
        rightButton={{
          label: 'Right',
          value: 'right',
        }}
      />,
    ))

  test('renders correctly with right value', () =>
    shouldMatchEmotionSnapshot(
      <SwitchButton
        name="test"
        value="right"
        onChange={() => {}}
        leftButton={{
          label: 'Left',
          value: 'left',
        }}
        rightButton={{
          label: 'Right',
          value: 'right',
        }}
      />,
    ))

  test('renders with tooltip', () =>
    shouldMatchEmotionSnapshot(
      <SwitchButton
        name="test"
        onChange={() => {}}
        leftButton={{
          label: 'Left',
          value: 'left',
        }}
        rightButton={{
          label: 'Right',
          value: 'right',
        }}
        tooltip="This is a tooltip"
      />,
    ))

  test('renders with on change', () => {
    const onChange = jest.fn()

    const node = renderWithTheme(
      <SwitchButton
        name="test"
        onChange={onChange}
        leftButton={{
          label: 'Left',
          value: 'left',
        }}
        rightButton={{
          label: 'Right',
          value: 'right',
        }}
        tooltip="This is a tooltip"
      />,
    )

    const input = node.getAllByRole('radio', {
      hidden: true,
    })

    userEvent.click(input[1])
  })
})
