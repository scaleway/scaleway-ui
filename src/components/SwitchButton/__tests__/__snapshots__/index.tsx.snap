// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchButton renders correctly 1`] = `
<DocumentFragment>
  .cache-1t6qoky-StyledSwitch-segment {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-style: solid;
  border-color: #d4dae7;
  border-width: 1px;
  padding: 16px;
  -webkit-transition: color 0.2s,border-color 0.2s,box-shadow 0.2s;
  transition: color 0.2s,border-color 0.2s,box-shadow 0.2s;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  color: #b2b6c3;
  font-size: 16px;
  line-height: 22px;
  position: relative;
  font-weight: 500;
  cursor: pointer;
}

.cache-1t6qoky-StyledSwitch-segment[aria-checked='true'] {
  cursor: auto;
  color: #390171;
  border-color: #390171;
}

.cache-1t6qoky-StyledSwitch-segment[aria-disabled='true'] {
  cursor: not-allowed;
  color: #d4dae7;
  background-color: #f6f6f8;
  border-color: #fcfcfd;
  pointer-events: none;
}

.cache-1t6qoky-StyledSwitch-segment[aria-disabled='true'][aria-checked='true'] {
  color: #d4dae7;
  border-color: #fcfcfd;
}

.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):hover,
.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):focus {
  border-color: #390171;
}

.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):hover {
  box-shadow: 0px 4px 16px 4px #EEEEFFCC;
}

.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):focus {
  box-shadow: 0px 0px 0px 3px #4F059940;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'] {
  font-size: 14;
  height: 40px;
  -webkit-transition: none;
  transition: none;
  border-radius: 4px;
  box-shadow: none;
  border-color: transparent;
  color: #4a4f62;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:hover,
.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:focus {
  border-color: transparent;
  box-shadow: none;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'][aria-checked='true'] {
  background-color: #4f0599;
  color: #ffffff;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'][aria-disabled='true'] {
  color: #d4dae7;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'][aria-disabled='true'][aria-checked='true'] {
  color: #d4dae7;
  background-color: #f6f6f8;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:not([aria-checked='true']):not([aria-disabled='true']):hover,
.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:not([aria-checked='true']):not([aria-disabled='true']):focus {
  border-color: transparent;
  box-shadow: none;
}

.cache-18s3dma-StyledRadio {
  position: absolute;
  opacity: 0.01;
}

<label
    aria-checked="false"
    aria-disabled="false"
    class="e16krxq01 cache-1t6qoky-StyledSwitch-segment"
    for="radio-choice"
  >
    Choice
    <input
      aria-checked="false"
      class="cache-18s3dma-StyledRadio e16krxq00"
      id="radio-choice"
      name="radio"
      tabindex="0"
      type="radio"
      value="choice"
    />
  </label>
</DocumentFragment>
`;

exports[`SwitchButton renders correctly when disabled 1`] = `
<DocumentFragment>
  .cache-1t6qoky-StyledSwitch-segment {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-style: solid;
  border-color: #d4dae7;
  border-width: 1px;
  padding: 16px;
  -webkit-transition: color 0.2s,border-color 0.2s,box-shadow 0.2s;
  transition: color 0.2s,border-color 0.2s,box-shadow 0.2s;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  color: #b2b6c3;
  font-size: 16px;
  line-height: 22px;
  position: relative;
  font-weight: 500;
  cursor: pointer;
}

.cache-1t6qoky-StyledSwitch-segment[aria-checked='true'] {
  cursor: auto;
  color: #390171;
  border-color: #390171;
}

.cache-1t6qoky-StyledSwitch-segment[aria-disabled='true'] {
  cursor: not-allowed;
  color: #d4dae7;
  background-color: #f6f6f8;
  border-color: #fcfcfd;
  pointer-events: none;
}

.cache-1t6qoky-StyledSwitch-segment[aria-disabled='true'][aria-checked='true'] {
  color: #d4dae7;
  border-color: #fcfcfd;
}

.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):hover,
.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):focus {
  border-color: #390171;
}

.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):hover {
  box-shadow: 0px 4px 16px 4px #EEEEFFCC;
}

.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):focus {
  box-shadow: 0px 0px 0px 3px #4F059940;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'] {
  font-size: 14;
  height: 40px;
  -webkit-transition: none;
  transition: none;
  border-radius: 4px;
  box-shadow: none;
  border-color: transparent;
  color: #4a4f62;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:hover,
.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:focus {
  border-color: transparent;
  box-shadow: none;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'][aria-checked='true'] {
  background-color: #4f0599;
  color: #ffffff;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'][aria-disabled='true'] {
  color: #d4dae7;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'][aria-disabled='true'][aria-checked='true'] {
  color: #d4dae7;
  background-color: #f6f6f8;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:not([aria-checked='true']):not([aria-disabled='true']):hover,
.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:not([aria-checked='true']):not([aria-disabled='true']):focus {
  border-color: transparent;
  box-shadow: none;
}

.cache-18s3dma-StyledRadio {
  position: absolute;
  opacity: 0.01;
}

<label
    aria-checked="false"
    aria-disabled="true"
    class="e16krxq01 cache-1t6qoky-StyledSwitch-segment"
    for="radio-choice"
  >
    Choice
    <input
      aria-checked="false"
      aria-disabled="true"
      class="cache-18s3dma-StyledRadio e16krxq00"
      disabled=""
      id="radio-choice"
      name="radio"
      style="pointer-events: none;"
      type="radio"
      value="choice"
    />
  </label>
</DocumentFragment>
`;

exports[`SwitchButton renders correctly with complex child 1`] = `
<DocumentFragment>
  .cache-1t6qoky-StyledSwitch-segment {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-style: solid;
  border-color: #d4dae7;
  border-width: 1px;
  padding: 16px;
  -webkit-transition: color 0.2s,border-color 0.2s,box-shadow 0.2s;
  transition: color 0.2s,border-color 0.2s,box-shadow 0.2s;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  color: #b2b6c3;
  font-size: 16px;
  line-height: 22px;
  position: relative;
  font-weight: 500;
  cursor: pointer;
}

.cache-1t6qoky-StyledSwitch-segment[aria-checked='true'] {
  cursor: auto;
  color: #390171;
  border-color: #390171;
}

.cache-1t6qoky-StyledSwitch-segment[aria-disabled='true'] {
  cursor: not-allowed;
  color: #d4dae7;
  background-color: #f6f6f8;
  border-color: #fcfcfd;
  pointer-events: none;
}

.cache-1t6qoky-StyledSwitch-segment[aria-disabled='true'][aria-checked='true'] {
  color: #d4dae7;
  border-color: #fcfcfd;
}

.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):hover,
.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):focus {
  border-color: #390171;
}

.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):hover {
  box-shadow: 0px 4px 16px 4px #EEEEFFCC;
}

.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):focus {
  box-shadow: 0px 0px 0px 3px #4F059940;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'] {
  font-size: 14;
  height: 40px;
  -webkit-transition: none;
  transition: none;
  border-radius: 4px;
  box-shadow: none;
  border-color: transparent;
  color: #4a4f62;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:hover,
.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:focus {
  border-color: transparent;
  box-shadow: none;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'][aria-checked='true'] {
  background-color: #4f0599;
  color: #ffffff;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'][aria-disabled='true'] {
  color: #d4dae7;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'][aria-disabled='true'][aria-checked='true'] {
  color: #d4dae7;
  background-color: #f6f6f8;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:not([aria-checked='true']):not([aria-disabled='true']):hover,
.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:not([aria-checked='true']):not([aria-disabled='true']):focus {
  border-color: transparent;
  box-shadow: none;
}

.cache-18s3dma-StyledRadio {
  position: absolute;
  opacity: 0.01;
}

<label
    aria-checked="true"
    aria-disabled="true"
    class="e16krxq01 cache-1t6qoky-StyledSwitch-segment"
    for="radio-choice"
  >
    true true
    <input
      aria-checked="true"
      aria-disabled="true"
      checked=""
      class="cache-18s3dma-StyledRadio e16krxq00"
      disabled=""
      id="radio-choice"
      name="radio"
      style="pointer-events: none;"
      type="radio"
      value="choice"
    />
  </label>
</DocumentFragment>
`;

exports[`SwitchButton renders correctly with variant segment 1`] = `
<DocumentFragment>
  .cache-1t6qoky-StyledSwitch-segment {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-style: solid;
  border-color: #d4dae7;
  border-width: 1px;
  padding: 16px;
  -webkit-transition: color 0.2s,border-color 0.2s,box-shadow 0.2s;
  transition: color 0.2s,border-color 0.2s,box-shadow 0.2s;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  touch-action: manipulation;
  color: #b2b6c3;
  font-size: 16px;
  line-height: 22px;
  position: relative;
  font-weight: 500;
  cursor: pointer;
}

.cache-1t6qoky-StyledSwitch-segment[aria-checked='true'] {
  cursor: auto;
  color: #390171;
  border-color: #390171;
}

.cache-1t6qoky-StyledSwitch-segment[aria-disabled='true'] {
  cursor: not-allowed;
  color: #d4dae7;
  background-color: #f6f6f8;
  border-color: #fcfcfd;
  pointer-events: none;
}

.cache-1t6qoky-StyledSwitch-segment[aria-disabled='true'][aria-checked='true'] {
  color: #d4dae7;
  border-color: #fcfcfd;
}

.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):hover,
.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):focus {
  border-color: #390171;
}

.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):hover {
  box-shadow: 0px 4px 16px 4px #EEEEFFCC;
}

.cache-1t6qoky-StyledSwitch-segment:not([aria-checked='true']):not([aria-disabled='true']):focus {
  box-shadow: 0px 0px 0px 3px #4F059940;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'] {
  font-size: 14;
  height: 40px;
  -webkit-transition: none;
  transition: none;
  border-radius: 4px;
  box-shadow: none;
  border-color: transparent;
  color: #4a4f62;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:hover,
.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:focus {
  border-color: transparent;
  box-shadow: none;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'][aria-checked='true'] {
  background-color: #4f0599;
  color: #ffffff;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'][aria-disabled='true'] {
  color: #d4dae7;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment'][aria-disabled='true'][aria-checked='true'] {
  color: #d4dae7;
  background-color: #f6f6f8;
}

.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:not([aria-checked='true']):not([aria-disabled='true']):hover,
.cache-1t6qoky-StyledSwitch-segment[data-variant='segment']:not([aria-checked='true']):not([aria-disabled='true']):focus {
  border-color: transparent;
  box-shadow: none;
}

.cache-18s3dma-StyledRadio {
  position: absolute;
  opacity: 0.01;
}

<label
    aria-checked="false"
    aria-disabled="false"
    class="e16krxq01 cache-1t6qoky-StyledSwitch-segment"
    data-variant="segment"
    for="radio-choice"
  >
    Choice
    <input
      aria-checked="false"
      class="cache-18s3dma-StyledRadio e16krxq00"
      id="radio-choice"
      name="radio"
      tabindex="0"
      type="radio"
      value="choice"
    />
  </label>
</DocumentFragment>
`;
